docs/CODE_GUIDELINES.md

```diff
@@ -764,4 +850,44 @@ Try to avoid using C macros. In many cases, they can be easily substituted with
 
 Prefer `constexpr` over `const` for constants when possible. Try to mark functions `constexpr` when reasonable.
 
+### 12.11. `std::string` vs `std::string_view`
+
+Prefer `std::string_view` over `std::string` when possible, for example constants variables, or method arguments, where in this last case it is not required declare the argument as reference or const, but of course depends on the type of implementation you want to achieve.
```

Comment:
```suggestion
Prefer `std::string_view` over `std::string` when reasonable. Good examples are constants or method arguments. In the last case, it is not required declare the argument as reference or const, since string views are immutable by definition. A bad example is when you need a NUL-terminated C string, e.g. to interact with a C API. `std::string_view` does not offer a `c_str()` function like `std::string` does.
```
I don't think the "depends on the type of implementation you want to achieve" adds much value; what exactly do you mean by that?

Reply:
> "depends on the type of implementation you want to achieve"

because you can change the view of the string_view used
for example by using `remove_prefix` you modify the view of the string_view
if you declare the stringview as const, you cannot use methods that modify the view, like mentioned above, so depends on what you want achieve, but most of cases add "const" is not needed

> since string views are immutable by definition

not sure if this sentence could be so good maybe reword it,
because you can derivate one view from another string_view by using methods, the value will be different, despite the source data are the same, like my example with `substr`

so maybe reformulate as:
_since the data source of string views are immutable by definition_

Reply:
> `std::string_view` does not offer a `c_str()` function like `std::string` does.

the similar way of  `c_str()` to `std::string_view` is `data()`
with the difference that you obtain the data not filtered by the view, so directly the source
we can explain this difference instead

Reply:
done all changes, try to see if better now

Reply:
> if you declare the stringview as const, you cannot use methods that modify the view, like mentioned above, so depends on what you want achieve, but most of cases add "const" is not needed

This makes sense for a local variable and such, but re-reading the text now I see that it is specifically meant to apply for method arguments. Do we ever qualify non-reference non-pointer method arguments as `const`? :thinking:  AFAIK, this is discouraged (in some guidelines at least) since it makes an internal concern of the method part of the API.

> the similar way of c_str() to std::string_view is data()

Yeah that much is clear, but it won't help if you need  a C string. That said, I think your new wording is fine. Thanks :)